spring-batch-integration/src/main/java/org/springframework/batch/integration/async/AsyncItemWriter.java

```diff
@@ -81,8 +81,10 @@ public void write(List<? extends Future<T>> items) throws Exception {
 				}
 			}
 		}
-		
-		delegate.write(list);
+
+		if (!list.isEmpty()) {

```

Comment:
What to do with empty chunks is a decision to be made by the delegate writer, not the asynchronous item writer. A concrete example is that when the delegate logs a warning about receiving an empty chunk, with this change, the log message would disappear. The role of the async item writer is to unwrap `Future`s objects and pass them to the delegate writer without any modification or additional logic. I will omit this change from the PR.